### PR TITLE
fix(adapter-node-http): Remove module monkey patching on disconnect

### PR DESCRIPTION
**Pull request by woodenconsulting**
## Description

Fixes a memory leak with Jest due to native module monkey patching. More details in the issue I filed [here](https://github.com/Netflix/pollyjs/issues/370). Note that I did not add any tests; implementing tests to assert the existence of a memory leak is not simple, but the changes here are. This is easily demonstrated in any jest test suite using polly with the node-http adapter by running the following command:

```
jest --coverage --verbose --colors --logHeapUsage --runInBand --detectLeaks
```

## Motivation and Context

Memory leaks are bad :)

## Types of Changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] I have added tests to cover my changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] My code follows the code style of this project.
- [x] My commits and the title of this PR follow the [Conventional Commits Specification](https://www.conventionalcommits.org).
- [x] I have read the [contributing guidelines](https://github.com/Netflix/pollyjs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: packages/@pollyjs/adapter-node-http/src/index.js [hunk 1]
```diff
@@ -120,6 +121,11 @@ export default class HttpAdapter extends Adapter {
       const module = modules[moduleName];
       const { request, get, globalAgent } = module;
 
+      this[moduleName] = {
```
Comment (jasonmit) [inline packages/@pollyjs/adapter-node-http/src/index.js:124]: nit: I'll probably end up renaming this but don't feel like blocking the PR on it :)

**Comment by jasonmit** [comment]
Thanks for catching & fixing the memory leak - they're rarely fun to track down so much appreciated 🙏

**Comment by jasonmit** [comment]
Fixes #370

**Comment by offirgolan** [comment]
Released with `v5.0.2`

**Pull request merged**